lib/ansible/modules/cloud/podman/podman_container.py

```diff
@@ -0,0 +1,1196 @@
+            for vol in params['volumes_from']:
+                cmd += ['--volumes-from', vol]
+        if params.get('workdir'):
+            cmd += ['--workdir', params['workdir']]
+        # Add your own args for podman command
+        if params.get('cmd_args'):
+            cmd += params['cmd_args']
+        cmd.append(params['image'])
+        if params.get('command'):
+            cmd += params['command'].split()
+        return cmd
+
+
+def ensure_image_exists(module, image):
+    rc, out, err = run_podman_command(module, args=['image', 'exists', image], ignore_errors=True)
+    if rc == 0:
+        return
+    rc, out, err = run_podman_command(module, args=['image', 'pull', image], ignore_errors=True)
+    if rc != 0:
+        module.fail_json(msg="Can't pull image %s" % image, stdout=out, stderr=err)
+    return True
+
+
+class PodmanContainer:
+    def __init__(self, module, name, params):
+
+        super(PodmanContainer, self).__init__()
+        self.module = module
+        self.name = name
+        self.info = self.get_info()
+
+    @property
+    def exists(self):
+        return bool(self.info != {})
+
+    @property
+    def different(self):
+        # TODO(sshnaidm): implement difference calculation between input vars
+        # and current container to understand if we need to recreate it
+        return True
+
+    @property
+    def running(self):
+        return self.exists and self.info['State']['Running']
+
+    @property
+    def stopped(self):
+        return self.exists and not self.info['State']['Running']
+
+    def get_info(self):
+        rc, out, err = run_podman_command(
+            module=self.module,
+            args=['container', 'inspect', self.name], ignore_errors=True)
+        return json.loads(out)[0] if rc == 0 else {}
+
+    def _perform_action(self, action):
+        command = construct_command_from_params(action, self.module.params)
+        self.module.log("PODMAN-DEBUG: " + " ".join(command))
+        rc, out, err = run_podman_command(
+            module=self.module,
+            args=['container'] + command,
+            ignore_errors=True)
+        if rc != 0:
+            self.module.fail_json(
+                msg="Can't %s container %s" % (action, self.name),
+                stdout=out, stderr=err)
+
+    def run(self):
+        self._perform_action('run')
+
+    def delete(self):
+        self._perform_action('delete')
+
+    def stop(self):
+        self._perform_action('stop')
+
+    def start(self):
+        self._perform_action('start')
+
+    def create(self):
+        self._perform_action('create')
+
+    def recreate(self):
+        self.delete()
+        self.run()
+
+    def restart(self):
+        self.stop()
+        self.run()
+
+
+class PodmanManager:
+
+    def __init__(self, module, results):
+
+        super(PodmanManager, self).__init__()
+
+        self.module = module
+        self.results = results
+        self.name = self.module.params.get('name')
```

Comment:
Don't use `.get()` method on things which are guaranteed to exist.
```suggestion
        self.name = self.module.params['name']
```

Reply:
sorry, removed all .get (I hope)